### ISSUE TITLE
Please separate code and autogenerated data

### ISSUE THREAD

**Issue by not-my-profile** [AlexWaygood/typeshed-stats#56]
I think `examples` and `stats_website` should both live either in another branch or another repository, so that updates to them do not clutter up the commit history of the `main` branch of this repository.

**Comment by AlexWaygood** [comment]
I agree that this isn't ideal. However, I'm inclined to keep things as they are for now.

Firstly, the current implementation is simple, works pretty well AFAICS, and fits my brain very well.

Secondly, not all files in the `stats_website` directory are autogenerated: `big_logo.png`, `index.md`, `logo.png`, `serialize.md` and `stats-csv.md` are all nongenerated.

Thirdly, at some point it might be nice to convert some of the stuff in `scripts/regenerate.py` into an actual plugin for `mkdocs` or `mkdocstrings` (Not one uploaded to PyPI, just one that would live in this repo). Not sure it makes much sense to make changes here until I've looked at that properly.

So: "maybe, but not right now".

**Comment by not-my-profile** [comment]
I'd love to contribute some improvements to the website, but I don't want to touch a repository that's actively using autogenerated commits ... but it's not a pressing matter, I am quite busy with other projects anyway.

Ad 2: Not a problem files can easily be copied.
Ad 3: I don't really see why this project should generate Markdown at all, or use a static-site generator. I think it makes more sense to directly generate HTML.

**Comment by AlexWaygood** [comment]
> Ad 3: I don't really see why this project should generate Markdown at all, or use a static-site generator. I think it makes more sense to directly generate HTML.

Yeah, quite possibly. Again, it was sort of a "hey, this works! Let's do it!" kind of thing. If it's useful for other people then that's awesome, but this is basically a hobby project for me first and foremost. I'm mostly here to have fun 😀

**Comment by not-my-profile** [comment]
Oh I just realized that the website also includes the API documentation, which you generate with mkdocs? That's quite nifty :), so using mkdocs does probably make sense.

**Comment by AlexWaygood** [comment]
> I'd love to contribute some improvements to the website, but I don't want to touch a repository that's actively using autogenerated commits ...

Separate to the principle of the thing (which I agree you're correct on, the autogenerated data should indeed be more cleanly separated from the code in an ideal world) — what exactly is your concern here, out of curiosity? Security issues?

**Comment by AlexWaygood** [comment]
You'll be pleased to know that, following #82, none of the MarkDown files in the `stats_website/` directory are autogenerated anymore (the generation is now done transparently as part of the mkdocs build, using the mkdocs-macros plugin). The `examples/` directory and `stats_website/stats_as_csv.csv` remain generated, for now.

**Comment by not-my-profile** [comment]
>what exactly is your concern here, out of curiosity?

I like to put effort into my commits to make the git log as readable as possible and don't really feel compelled to do so when the repository is actively using autogenerated commits.

Sidenote: [git log also doesn't have an easy way to exclude an author](https://stackoverflow.com/questions/6889830/equivalence-of-git-log-exclude-author) but even if it did, many people probably browse the log via the GitHub web interface which also doesn't have this functionality.

**Comment by AlexWaygood** [comment]
> I like to put effort into my commits to make the git log as readable as possible and don't really feel compelled to do so when the repository is actively using autogenerated commits.

Very fair, I get that! There will be fewer autogenerated commits now. I'm continuing to work on getting them down to 0.
